tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virtual/VirtualThreadServlet.java

```diff
@@ -0,0 +1,468 @@
+            Object result = virtualManagedExecutorDD
+                    .invokeAny(List.of(new LookupActionCaptureThread(null, "java:app/concurrent/ManagedExecutorDDVirtual"),
+                            new LookupActionCaptureThread(null, "java:app/concurrent/ManagedExecutorDDVirtual")));
+
+            assertNotNull(result);
+            if (result instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result);
+            assertTrue(isVirtual((Thread) result));
+        });
+
+    }
+
+    public void testPlatformScheduledExecutor() throws Exception {
+        ManagedScheduledExecutorService platformManagedScheduledExecutorAnno = InitialContext
+                .doLookup("java:app/concurrent/ManagedScheduledExecutorAnnoPlatform");
+        ManagedScheduledExecutorService platformManagedScheduledExecutorDD = InitialContext
+                .doLookup("java:app/concurrent/ManagedScheduledExecutorDDPlatform");
+        
+        assertNotNull(platformManagedScheduledExecutorAnno);
+        assertNotNull(platformManagedScheduledExecutorDD);
+
+        Future<Thread> future;
+
+        future = platformManagedScheduledExecutorAnno.submit(Thread::currentThread);
+        assertFalse(isVirtual(future.get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+
+        future = platformManagedScheduledExecutorDD.submit(Thread::currentThread);
+        assertFalse(isVirtual(future.get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+    }
+
+    public void testVirtualScheduledExecutor() throws Exception {
+        ManagedScheduledExecutorService virtualManagedScheduledExecutorAnno = InitialContext
+                .doLookup("java:app/concurrent/ManagedScheduledExecutorAnnoVirtual");
+        ManagedScheduledExecutorService virtualManagedScheduledExecutorDD = InitialContext
+                .doLookup("java:app/concurrent/ManagedScheduledExecutorDDVirtual");
+
+        assertNotNull(virtualManagedScheduledExecutorAnno);
+        assertNotNull(virtualManagedScheduledExecutorDD);
+        
+        Thread annoThread = virtualManagedScheduledExecutorAnno.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        Thread ddThread = virtualManagedScheduledExecutorDD.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        
+        if (VERSION == 17) { //TODO remove when Concurrency API supports only 21+
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(annoThread), "Should be impossible to get a virtual thread on Java 17");
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(ddThread), "Should be impossible to get a virtual thread on Java 17");
+            return;
+        }
+
+        // Java 21+
+        assertEquals(isVirtual(annoThread), isVirtual(ddThread));
```

Comment:
Similar to first comment. This is not a requirement, but very unlikely it would not be the case.